File: DAA/MatrixMul.java
public class MatrixMul {
    public static void main(String[] args) {
        int[][] a = { { 52, 73 }, { 53, 69 } };
        int[][] b = { { 02, 56 }, { 62, 66 } };
        int[][] c;

        System.out.println("Matrix A: ");
        display(a);

        System.out.println("Matrix B: ");
        display(b);

        c = MatrixMultiply(a, b);

        System.out.println("Resultant Matrix C: ");
        display(c);
    }

    static int[][] MatrixMultiply(int[][] a, int[][] b) {
        int n = a.length;
        int[][] c = new int[n][n];
        int i, j, k;

        for (i = 0; i < n; i++) {
            for (j = 0; j < n; j++) {
                for (k = 0; k < n; k++) {
                    c[i][j] += a[i][k] * b[k][j];
                }
            }
        }

        return c;
    }

    static void display(int[][] a) {
        for (int i = 0; i < a.length; i++) {
            for (int j = 0; j < a[i].length; j++) {
                System.out.print(a[i][j] + " ");
            }
            System.out.println("");
        }
    }
}

File: DAA/MatrixMul.c
#include <stdio.h>

void display(int matrix[2][2]) {
    for (int i = 0; i < 2; i++) {
        for (int j = 0; j < 2; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
}

void matrixMultiply(int a[2][2], int b[2][2], int c[2][2]) {
    for (int i = 0; i < 2; i++) {
        for (int j = 0; j < 2; j++) {
            c[i][j] = 0;
            for (int k = 0; k < 2; k++) {
                c[i][j] += a[i][k] * b[k][j];
            }
        }
    }
}

int main() {
    int a[2][2] = {{52, 73}, {53, 69}};
    int b[2][2] = {{02, 56}, {62, 66}};
    int c[2][2];

    printf("Matrix A: \n");
    display(a);

    printf("Matrix B: \n");
    display(b);

    matrixMultiply(a, b, c);

    printf("Resultant Matrix C: \n");
    display(c);

    return 0;
}
